Cmake update

lemenkov opened the pull request:
Small CMake updates stacked in my various branches.

juha-h commented:
Why threre is cmake version of sems.conf?  How does it matter regarding sems configuration, which make was used to build sems?

lemenkov commented:
> Why threre is cmake version of sems.conf? How does it matter regarding sems configuration, which make was used to build sems?

Agree -this looks uglier and uglier. I introduced it to redefine CONFIG_FILE, MOD_CFG_PATH, ANNOUNCE_PATH, PLUG_IN_PATH which is very simple in CMake if you're using templates.

I'll rewrite it and propose something different soon.

lemenkov commented:
> Why threre is cmake version of sems.conf? How does it matter regarding sems configuration, which make was used to build sems?

* Removed all CMake-related templates.
* Rebased on top of current master
